release-notes/WIP-cabal-install-3.12.0.0.md

```diff
@@ -0,0 +1,235 @@
+☞ N.B.: this is a WIP release-note file for 3.12.x.0 cabal-install.
+
+cabal-install and cabal-install-solver 3.12.0.0 changelog and release notes
```

Comment:
I'd be even more paranoid and rename 3.12.0.0 to 3.12.1.0 here and in the name of the file (or to 3.12.x.0 if you insist). Your call, though.